haversine/__init__.py

```diff
@@ -38,3 +38,40 @@ def haversine(point1, point2, miles=False, nautical_miles=False):
     lng = lng2 - lng1
     d = sin(lat * 0.5) ** 2 + cos(lat1) * cos(lat2) * sin(lng * 0.5) ** 2
     return 2 * avg_earth_radius * asin(sqrt(d))
+
+
+def inverse_haversine(point, distance, direction, miles=False, nautical_miles=False):
+
+    if miles:
+        avg_earth_radius = AVG_EARTH_RADIUS_MI
+    elif nautical_miles:
+        avg_earth_radius = AVG_EARTH_RADIUS_NMI
+    else:
+        avg_earth_radius = AVG_EARTH_RADIUS_KM
+
+    lat1, lng1 = point
+    lat1, lng1 = map(radians, (lat1, lng1))
+    d = sin(distance / (2 * avg_earth_radius)) ** 2
+
+    if direction == "north":
```

Comment:
This way your formula should be simple if I remember my math right:

```py
lat2 = lat1 + distance * sin(angle)
lon2 = lon2 + distance * cos(angle)
```

You may have to adapt if you are North-East, South-East, South-West or North-West by subtract instead of adding.

Reply:
This math only works when on a plane, not a surface with curvature like the earth, at least from my understanding. I've been doing research on this topic and the solution to a problem like this is a direct geodesic solution. I'm looking into how one might implement this in python. See http://en.wikipedia.org/wiki/Geodesics_on_an_ellipsoid and https://en.wikipedia.org/wiki/Great-circle_distance

Reply:
You are totally right, I assumed that the approximation would work in "small" distances, but if you have a better implementation, let's do this, it will be way better ;)